### ISSUE TITLE
Only bounding box but no segmentation mask in prediction

### ISSUE THREAD

**Issue by Hanson13** [matterport/Mask_RCNN#1405]
I used the following project to train on my own dataset and got a trained weight successfully. 
https://github.com/gustavz/Mobile_Mask_RCNN
However, when I used the weight to inference on my dataset, the results only contained the bounding box of detection but no mask of segmentation. Does anyone have any idea why it happened? Thanks a lot!

**Comment by monjurulkarim** [comment]
You should check the display_instance() function in visualize.py file.

**Comment by Hanson13** [comment]
@monjurulkarim Thank you for reply. I have checked the display_instance(), and it worked normally. However, I find that the prediction result actually contained the segmentation mask, but the mask is the same as the bounding box, whose shape is a rectangle instead of a polygon. I am sure the label of training set contained the information of polygon segmentation. I doubt that the model did not read the segmentation label when training, but don't know how to debug. Can anyone tell where the model reads the segmentation label in the code?

**Comment by monjurulkarim** [comment]
@Hanson13 You should check whether the training data contains rectangle bounding box or polygon mask.
For loading the dataset, there should be a function load_data() or something like that. And for loading the polygon mask, there is a function load_mask(). I think you can debug from there to know whether it is reading the both x and y coordinates of the polygons correctly or not.

**Comment by eddy9254** [comment]
@Hanson13  
Hi, Thank you for making this discussion. by the way, did you solve the problems?? 
because i'm having same problems with you.  
when i used the my trained weight to inference on my dataset, the results contains both bounding boxs and masks but the masks does not filled with color just like the below picture.
it seems like the result shows only the area of masks without including segmentation.
i checked my display_instance() function in visualize.py file but nothing was changed.

![image](https://user-images.githubusercontent.com/37107259/56340062-a8e06d80-61eb-11e9-905b-2ac7098b79ad.png)
